### PR TITLE
Fix test nanny timeout

### PR DESCRIPTION
**Pull request by fjetter**
This should be fixing `distributed.tests.test_nanny.test_nanny_timeout` which appears to be caused by a race condition between startup and kill.

That condition is now tested in a dedicated test `test_worker_start_exception_while_killing`

### CODE REVIEW

File: distributed/tests/test_nanny.py [hunk 1]
```diff
@@ -582,6 +582,37 @@ async def test_worker_start_exception(s):
...
 
 
+@gen_cluster(nthreads=[])
+async def test_worker_start_exception_while_killing(s):
+    nanny = Nanny(s.address, worker_class=BrokenWorker)
+
+    async def try_to_kill_nanny():
+        while not nanny.process or nanny.process.status not in (
+            Status.starting,  # this is what we want
+            # Status.failed # we might've missed it already
```
Comment (hendrikmakait) [inline distributed/tests/test_nanny.py:592]:
```suggestion
```

**Review by hendrikmakait — APPROVED** [review]
Thanks, @fjetter!

**Comment by fjetter** [comment]
This still has a problem...

**Comment by hendrikmakait** [comment]
> This still has a problem...

Alright, just let me know once it's ready for a final review. CI looked good though 🙈

**Comment by fjetter** [comment]
@hendrikmakait  should be good to go now

**Pull request merged**